app/plugin_api.go

```diff
@@ -741,6 +741,10 @@ func (api *PluginAPI) KVList(page, perPage int) ([]string, *model.AppError) {
 	return api.app.ListPluginKeys(api.id, page, perPage)
 }
 
+func (api *PluginAPI) KVAtomicModify(key string, f func(initialValue []byte) ([]byte, error)) *model.AppError {
```

Comment:
Unfortunately, there is no way to submit a `func` via RPC. Hence, this function has to be a helper not a "full" RPC function.

Sorry for the drive by review.

Reply:
@hanzei Hello Ben, could you please tell me there I shod place that helper method? For now I just simply move it out from the Plugin API interface.

Reply:
There should be a `KVAtomicModify` function in https://github.com/mattermost/mattermost-server/blob/master/plugin/helpers.go and the implementation should go into https://github.com/mattermost/mattermost-server/blob/master/plugin/helpers_kv.go.

@igomonov88 Does this make it more clear?

Reply:
@igomonov88 Let me know if you need help with making this a helper function.

Reply:
@hanzei Hi, here is requested updates I guess. Thank you guys for waiting.